drivers/stepper/gpio_stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,201 @@
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct gpio_stepper_motor_controller_data *data =
+		CONTAINER_OF(dwork, struct gpio_stepper_motor_controller_data, stepper_dwork);
+	LOG_DBG("Moving Motor from %d to %d with delay %d ", data->actual_position,
+		data->target_position, data->delay_in_us);
+	static int32_t step_count = 1;
+
+	if (data->actual_position != data->target_position) {
+		if (data->actual_position > data->target_position) {
+			stepper_motor_move_with_4_gpios(data->gpio_in_pins, step_count++);
+			data->actual_position--;
+		} else if (data->actual_position < data->target_position) {
+			stepper_motor_move_with_4_gpios(data->gpio_in_pins, step_count--);
+			data->actual_position++;
+		}
+		data->is_moving = true;
+		k_work_reschedule(&data->stepper_dwork, K_USEC(data->delay_in_us));
+	} else {
+		data->is_moving = false;
+	}
+}
+
+static int32_t gpio_stepper_move(const struct device *dev, int32_t micro_steps)
+{
+	struct gpio_stepper_motor_controller_data *data =
+		(struct gpio_stepper_motor_controller_data *)dev->data;
+	data->target_position = data->actual_position + micro_steps;
+	k_work_reschedule(&data->stepper_dwork, K_NO_WAIT);
+	return 0;
+}
+
+static int32_t gpio_stepper_set_actual_position(const struct device *dev, int32_t position)
+{
+	struct gpio_stepper_motor_controller_data *data =
+		(struct gpio_stepper_motor_controller_data *)dev->data;
+	data->actual_position = position;
+	return 0;
+}
+
+static int32_t gpio_stepper_get_actual_position(const struct device *dev, int32_t *position)
+{
+	struct gpio_stepper_motor_controller_data *data =
+		(struct gpio_stepper_motor_controller_data *)dev->data;
+	*position = data->actual_position;
+	return 0;
+}
+
+static int32_t gpio_stepper_set_target_position(const struct device *dev, int32_t position)
+{
+	struct gpio_stepper_motor_controller_data *data =
+		(struct gpio_stepper_motor_controller_data *)dev->data;
+	data->target_position = position;
+	k_work_reschedule(&data->stepper_dwork, K_NO_WAIT);
+	return 0;
+}
+
+static int gpio_stepper_motor_controller_init(const struct device *dev)
+{
+	const struct gpio_stepper_motor_controller_config *config =
+		(const struct gpio_stepper_motor_controller_config *)dev->config;
+	struct gpio_stepper_motor_controller_data *data =
+		(const struct gpio_stepper_motor_controller_data *)dev->data;
+	LOG_DBG("Initializing %s gpio_stepper_motor_controller with pins %d for motor with steps",
+		dev->name, config->number_in_pins);
+
+	for (uint8_t n_pin = 0; n_pin < config->number_in_pins; n_pin++) {
+		gpio_pin_configure_dt(&data->gpio_in_pins[n_pin], GPIO_OUTPUT_LOW);
+	}
+	k_work_init_delayable(&data->stepper_dwork, stepper_work_step_handler);
+
+	return 0;
+}
+
+static int gpio_stepper_is_moving(const struct device *dev, bool *is_moving)
+{
+	struct gpio_stepper_motor_controller_data *data =
+		(struct gpio_stepper_motor_controller_data *)dev->data;
+	*is_moving = data->is_moving;
```

Comment:
`is_moving` can be removed as it is tied 1-1 with the work item not being idle.
```
*is_moving = k_work_delayable_is_pending(&data->stepper_dwork);
```

Reply:
@bjarki-andreasen that's true for this driver, but that won't be true for all drivers (i.e. SPI-based drivers for controllers with integrated position and acceleration control).

Reply:
I know :) The change is just removing a redundant bool from this driver's data struck, the API is not altered :)